src/controllers/auth.ts

```diff
@@ -0,0 +1,108 @@
+// Dependencies
+import UUID from 'uuid/v4'
+
+// Local Dependencies
+import { Request, Response } from '../interfaces'
+import { isSignatureValid, generateJwtToken } from '../helpers'
+import { models } from '../db'
+
+const signup = async (req: Request, res: Response, next: Function) => {
+  const { Users } = models
+  const { publicAddress } = req.body
+
+  try {
+    const user = await Users.create({
+      publicAddress,
+      nonce: UUID().replace(/-/g, ''),
```

Comment:
who uses a nonce formatted like this?

Reply:
What do you mean? It’s just a random value that can be signed with a user’s private key to verify their identity during user authentication. Rather than generate a large random number, I’m just using a UUID and removing the ‘-‘.

Here’s our issue tracking the [technical spike on metamask auth](https://github.com/computablelabs/ld01/issues/16). I wrote up the learnings at the bottom of the issue description.